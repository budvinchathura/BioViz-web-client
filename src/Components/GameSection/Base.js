import React from 'react';
import {Avatar} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    avatar: {
        height: '20px',
        width: '20px',
        fontSize: '12px',

    },
    A: {
        color: 'white',
        backgroundColor: 'red',
    },
    C: {
        color: 'white',
        backgroundColor: 'blue',
    },
    G: {
        color: 'white',
        backgroundColor: 'purple',
    },
    T: {
        color: 'white',
        backgroundColor: 'green',
    },
    ga: {
        color: 'black',
        backgroundColor: 'black',
    },
    r: {
        color: 'black',
        backgroundColor: 'black',
    }

}));

export default function Base(props) {
    const classes = useStyles();
    const index = props.index;
    const base = props.base;
        
    return(
            <Avatar variant='square' id={index}
            className={`${classes.avatar} ${classes[base]}`} >
            {base}</Avatar>   
        );
}