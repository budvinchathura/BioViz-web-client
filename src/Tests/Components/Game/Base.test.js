import React from 'react';
React.useLayoutEffect = React.useEffect;
import renderer from 'react-test-renderer';
import Base from '../../../Components/GameSection/Base';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

describe('Base Component', ()=>{
    const store = mockStore({genomeType: 'DNA'});
    const testProp = {
        index: 4,
        base: 'G',
    };
    it('matches snapshot', ()=>{
        const tree = renderer.create(<Provider store={store}>
            <Base {...testProp} /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
