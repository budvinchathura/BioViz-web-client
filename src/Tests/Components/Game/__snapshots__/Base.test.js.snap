// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Component matches snapshot 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-1 DNA-G MuiAvatar-colorDefault"
  id={4}
>
  G
</div>
`;
